joomla/joomla-cms
# [4.2] Fix wrong data for calendar custom fields when user registration fail

Pull Request for Issue #39844

### Summary of Changes
Currently, if you use Calendar custom fields for user registration, if registration fail, there could be some issues with data of Calendar custom fields:
- Time of the selected date could be changed. It happens because the Calendar form field expect to receive input datetime from UTC timezone , but the provided value is from user timezone (filter is not applied yet)
- In some cases like when you use US language package, the Calendar field could not parse the selected date and result in fatal error

To address these issues, we can use same code which we are using on FormController https://github.com/joomla/joomla-cms/blob/4.2-dev/libraries/src/MVC/Controller/FormController.php#L839-L855 : Process filter for Calendar fields, store the filtered data back to request. 

### Testing Instructions
1. Follow instructions at https://github.com/joomla/joomla-cms/issues/39844 and confirm the issue
2. Apply patch, confirm that when registration fail, you do not get fatal error anymore. The date which you selected in the field remains as how it is.

### Actual result BEFORE applying this Pull Request
There is fatal error

### Expected result AFTER applying this Pull Request
No fatal error anymore


### Link to documentations
Please select:
